### PR TITLE
feat: improve DX for matrix struct

### PR DESCRIPTION
**Pull request by eddyxu**
* Implement a `From<FSL> for Matrix` to make conversion easier.
* Clean up float type trait

### CODE REVIEW

File: rust/lance-linalg/src/matrix.rs [hunk 1]
```diff
@@ -52,12 +55,25 @@ where
     pub transpose: bool,
 }
 
-impl<T: FloatArray> MatrixView<T>
+impl<T: ArrowFloatType> Clone for MatrixView<T>
 where
-    Standard: rand::distributions::Distribution<<T as FloatArray>::Native>,
+    Standard: rand::distributions::Distribution<<T as ArrowFloatType>::Native>,
```
Comment (westonpace) [inline rust/lance-linalg/src/matrix.rs:60]: Do you need the rand constraint for clone?

File: rust/lance-linalg/src/matrix.rs [hunk 2]
```diff
@@ -281,18 +297,59 @@ where
...
+            | (DataType::Float64, FloatType::Float64) => {}
+            _ => {
+                return Err(ArrowError::CastError(format!(
+                    "Can not convert from {}",
+                    value.value_type(),
+                )))
+            }
+        }
+        // Use downcast instead of `as_primitive()` to away from the type check.
+        // This is a bit
```
Comment (wjones127) [inline rust/lance-linalg/src/matrix.rs:320]: Incomplete comment?

**Review by wjones127 — APPROVED** [review]

File: rust/lance-linalg/src/matrix.rs (hunk 2 above)
Comment (eddyxu) [reply rust/lance-linalg/src/matrix.rs:320]: Fixed.

Comment (westonpace) [inline rust/lance-linalg/src/matrix.rs]: Maybe remove or clarify this comment.

**Review by westonpace — APPROVED** [review]
This looks good.

File: rust/lance-linalg/src/matrix.rs (hunk 1 above)
Comment (eddyxu) [reply rust/lance-linalg/src/matrix.rs:60]: Matrix has a random initialization method that needs this constraint. 

https://github.com/lancedb/lance/blob/27814c56fd99241ce4ded977896179746d27c74d/rust/lance-linalg/src/matrix.rs#L69

File: rust/lance-linalg/src/matrix.rs (hunk 2 above)
Comment (eddyxu) [reply rust/lance-linalg/src/matrix.rs]: Fixed

**Pull request merged**